TwitchLib.Client/TwitchClient.cs

```diff
@@ -1121,129 +1121,57 @@ private void JoinChannelTimeout(object sender, System.Timers.ElapsedEventArgs e)
         /// Handles the irc message.
         /// </summary>
         /// <param name="ircMessage">The irc message.</param>
-        private async Task HandleIrcMessageAsync(IrcMessage ircMessage)
+        private Task HandleIrcMessageAsync(IrcMessage ircMessage)
         {
-            if (ircMessage.ToString().StartsWith(":tmi.twitch.tv NOTICE * :Login authentication failed"))
-            {
-                if (OnIncorrectLogin != null)
-                    await OnIncorrectLogin.Invoke(this,
-                        new OnIncorrectLoginArgs
-                        {
-                            Exception = new ErrorLoggingInException(ircMessage.ToString(), TwitchUsername)
-                        });
-                return;
+            var rawMessage = ircMessage.ToString();
+            if (rawMessage.StartsWith(":tmi.twitch.tv NOTICE * :Login authentication failed"))
+            { 
+                // Does this need to fallback to UnaccountedFor?
+                return OnIncorrectLogin?.Invoke(this, new()
```

Comment:
I would be careful here. I don't think this change is correct.
`OnIncorrectLogin` is now `AsyncEventHandler`, which means that event is awaitable. But if you attempt to await null Task it will throw NullReferenceException. So you actually can't use `?.` operator here. That's why there was explicit `if` statement.